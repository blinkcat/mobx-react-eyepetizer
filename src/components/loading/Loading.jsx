import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import styles from './loading.css';

export default function Loading({ className }) {
	return (
		<div className={classNames(styles.root, { [className]: className })}>
			<div className={styles.out} />
			<div className={styles.in} />
		</div>
	);
}

Loading.propTypes = {
	className: PropTypes.string
};

Loading.defaultProps = {
	className: ''
};
