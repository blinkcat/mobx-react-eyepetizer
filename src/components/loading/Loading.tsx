import React from 'react';
import classNames from 'classnames';
import styles from './loading.css';

interface LoadingProp {
  className?: string;
}

export default function Loading({ className = '' }: LoadingProp) {
  return (
    <div className={classNames(styles.root, { [className]: className })}>
      <div className={styles.out} />
      <div className={styles.ining} />
    </div>
  );
}
