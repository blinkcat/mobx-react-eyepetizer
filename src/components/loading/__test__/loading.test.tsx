import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Loading from '../';

describe('Component Loading', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
