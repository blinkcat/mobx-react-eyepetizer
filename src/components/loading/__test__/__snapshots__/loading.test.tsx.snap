// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Loading renders correctly 1`] = `
<div
  className="loading"
>
  <div
    className="out"
  />
  <div
    className="in"
  />
  <style
    jsx={true}
  >
    
        .loading {
          background: #fff;
          height: 100%;
          opacity: 1;
          position: fixed;
          left: 0;
          transition: opacity 0.3s;
          width: 100%;
          z-index: 10000;
          div{
            background-size: cover;
            height: 48px;
            left: 50%;
            margin: -24px 0 0 -35px;
            position: absolute;
            top: 50%;
            width: 70px;
          }
          .out{
            background-image: url(
              //static.kaiyanapp.com/eyepetizer-web/assets/images/detail/loading_out@3x.444b89a3.png);
          }
          .in{
            animation: e 0.5s linear infinite;
            background-image: url(
              //static.kaiyanapp.com/eyepetizer-web/assets/images/detail/loading_in@3x.94a9c4b4.png);
          }
        }
        @keyframes e {
          0% {
            transform: rotate(0);
          }

          to {
            transform: rotate(1turn);
          }
        }
      
  </style>
</div>
`;
